modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,498 @@
+			origin: OriginFor<T>,
+			owner: T::AccountId,
+			account: T::AccountId,
+		) -> DispatchResult {
+			// Extract confirmation info from Origin.
+			let data = T::OracleOrigin::ensure_origin(origin)?;
+
+			// Checks whether oracle confirms or rejects the mint request
+			if data.get(0).ok_or(Error::<T>::BadOracleData)?.is_one() {
+				Self::accept_mint_request(owner, account)
+			} else {
+				Self::reject_mint_request(owner)
+			}
+		}
+
+		/// Requests to redeem an Account Token. Sends XCM message to the relaychain to transfer the
+		/// control of the account.
+		/// The NFT is taken into custodial by the module, and is not burned until confirmed by the
+		/// Oracle.
+		///
+		/// Params:
+		/// 	- `account`: Account ID of the Account Token
+		/// 	- `new_owner`: The owner of the proxy account to be transferred to.
+		#[pallet::weight(< T as Config >::WeightInfo::request_redeem())]
+		#[transactional]
+		pub fn request_redeem(origin: OriginFor<T>, account: T::AccountId, new_owner: T::AccountId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let nft_class_id = Self::nft_class_id();
+
+			// Obtain info about the account token.
+			let token_id = Self::minted_account(&account).ok_or(Error::<T>::AccountTokenNotFound)?;
+			let owner = T::NFTInterface::owner(&nft_class_id, &token_id).ok_or(Error::<T>::AccountTokenNotFound)?;
+
+			// Ensure that only the owner of the NFT can redeem.
+			ensure!(who == owner, Error::<T>::CallerUnauthorized);
+
+			// The XCM fee is burned.
+			T::Currency::withdraw(
+				&who,
+				T::XcmInterface::get_transfer_proxy_xcm_fee(),
+				WithdrawReasons::FEE,
+				KeepAlive,
+			)?;
+
+			// Send an XCM to relaychain to relinquish the control of the `account` to `new_owner`.
+			T::XcmInterface::transfer_proxy(account.clone(), new_owner.clone())?;
+
+			// Submit confirmation call to be serviced by foreign state oracle
+			let call: <T as Config>::Call = Call::<T>::confirm_redeem_account_token {
+				account: account.clone(),
+				new_owner: new_owner.clone(),
+			}
+			.into();
+			T::ForeignStateQuery::create_query(&who, call, None)?;
+
+			// Take custodial of the NFT token.
+			T::NFTInterface::transfer(&Self::nft_class_id(), &token_id, &Self::account_id())?;
+
+			Self::deposit_event(Event::RedeemRequested {
+				account,
+				owner,
+				token_id,
+				new_owner,
+			});
+			Ok(())
+		}
+
+		/// Confirm that the parachain account has relinquished the control of the account on the
+		/// relaychain to the `new_owner`. The NFT is burned and storage updated.
+		///
+		/// Only callable by the Oracle.
+		///
+		/// Params:
+		/// 	- `account`: Account ID of the Account Token
+		/// 	- `new_owner`: The owner of the proxy account to be transferred to.
+		#[pallet::weight(< T as Config >::WeightInfo::confirm_redeem_account_token())]
+		#[transactional]
+		pub fn confirm_redeem_account_token(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			new_owner: T::AccountId,
```

Comment:
I don't see any use of `new_owner`, do we need here?

Reply:
Hi i think it's used to deposit event. I think this can be removed